ext/spl/spl_iterators.c

```diff
@@ -1866,6 +1866,12 @@ PHP_METHOD(RegexIterator, accept)
 			}
 
 			result = php_pcre_replace_impl(intern->u.regex.pce, subject, ZSTR_VAL(subject), ZSTR_LEN(subject), replacement_str, -1, &count);
+			zend_string_release(replacement_str);
+			zend_string_release(subject);
```

Comment:
Seems like this PR was forgotten, but you cannot release subject here as it would cause a double free at the end of the function. This should only be released in the `!result` block.
Please test this with a non-interned subject string.